lake/writer.go

```diff
@@ -124,10 +124,13 @@ func (w *Writer) Close() error {
 }
 
 func (w *Writer) writeObject(object *data.Object, recs []zed.Value) error {
-	if !w.inputSorted {
+	var zr zio.Reader
+	if w.inputSorted {
+		zr = zbuf.NewArray(recs)
+	} else {
 		done := make(chan struct{})
 		go func() {
-			w.comparator.SortStable(recs)
+			zr = w.comparator.SortStableReader(recs)
```

Comment:
Just a though, and not to be addressed in this pr, but perhaps we should add a context to the comparator Sort* methods so we don't have to do things like this?

Reply:
You might be on to something here.